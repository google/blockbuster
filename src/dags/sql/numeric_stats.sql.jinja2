SELECT
  FORMAT_TIMESTAMP('%Y-%m-%d', TIMESTAMP_MILLIS(timeInMillis)) AS day,
  name,
  COUNT(*) AS fact_count,
  COUNT(SAFE_CAST(value AS INT64)) AS value_count,
  AVG(SAFE_CAST(value AS INT64)) AS average,
  STDDEV_SAMP(SAFE_CAST(value AS INT64)) AS stddev
FROM  `{{ fact_table }}`
WHERE name IN ({{ feature_columns|join(', ') }})
GROUP BY 1, 2;
