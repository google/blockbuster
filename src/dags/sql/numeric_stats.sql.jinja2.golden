SELECT
  FORMAT_TIMESTAMP('%Y-%m-%d', TIMESTAMP_MILLIS(timeInMillis)) AS day,
  name,
  COUNT(*) AS fact_count,
  COUNT(SAFE_CAST(value AS INT64)) AS value_count,
  AVG(SAFE_CAST(value AS INT64)) AS average,
  STDDEV_SAMP(SAFE_CAST(value AS INT64)) AS stddev
FROM  `project.dataset.table`
WHERE name IN (col1, col2, col3)
GROUP BY 1, 2;
